### PR TITLE
A few fixes & improvements

### PR DESCRIPTION
**Pull request by ColdFire87**
__Hey, thanks for writing this!__

I had a quick look through the code and did a few obvious fixes pointed out by my IDE (JetBrains Webstorm).

I also made some additional improvements:
- replaced deprecated package `@slack/client` with `@slack/web-api`
- added `serverless-iam-roles-per-function` plugin & removed unnecessary permissions
- added `serverless-pseudo-parameters` so the code can reference the AWS::AccountId in the ARNs
- removed `SLACK_TOKEN` from global env as it's retrieved in the lambda from SSM

The Jest tests were also failing (due to `console.log()` not found) so I edited `tsconfig.json`.

### CODE REVIEW

File: codebuild.ts [hunk 1]
```diff
@@ -198,15 +198,18 @@ const gitRevision = (event: CodeBuildEvent): string => {
     if (sourceVersion === undefined) {
       return 'unknown';
     }
-    // The location minus '.git'
-    const githubProjectUrl = event.detail[
-      'additional-information'
-    ].source.location.slice(0, '.git'.length);
+    const githubProjectUrl = event.detail['additional-information'].source.location;
```
Comment (joebowbeer) [inline codebuild.ts]:
```suggestion
    const githubProjectUrl = event.detail['additional-information'].source.location.slice(0, -('.git'.length));
```

File: package.json [hunk 2]
```diff
@@ -1,18 +1,20 @@
 {
-  "name": "codebuild-slack-notifier",
-  "version": "1.0.0",
-  "description": "Send CodeBuild events to Slack",
+  "name": "pipelines-slack-notifier",
```
Comment (joebowbeer) [inline package.json]: More than a minor change?

**Comment by joebowbeer** [comment]
Suggest adding tests for the gitRevision enhancement:

```typescript
describe('gitRevision', () => {
  it('gets unknown location from a state event', () => {
    expect(gitRevision(mockStateEvent)).toEqual('unknown');
  });

  it('gets GitHub location from a phase event', () => {
    expect(gitRevision(mockPhaseEvent)).toEqual(
      '<https://github.com/fishbrain/rutilus-api/commit/d5b0159a3ce05a95d7f16055dab3e0ae3cae6cff|d5b0159a3ce05a95d7f16055dab3e0ae3cae6cff>'
    );
  });
});
```

File: package.json (hunk 2 above)
Comment (ColdFire87) [reply package.json]: my bad, shouldn't have committed this change

**Comment by ColdFire87** [comment]
I'll add the tests when I get the chance (prob this weekend).

**Pull request merged**